### PR TITLE
py-cma: new package

### PR DESCRIPTION
**Pull request by LydDeb**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the homepage and version sha256.

File: var/spack/repos/builtin/packages/py-pycma/package.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+class PyPycma(PythonPackage):
+    """
+    Python implementation of CMA-ES Covariance Matrix Adaptation Evolution Str
+    ategy for non-linear numerical optimization in Python
+    """
+
+    homepage = "https://github.com/CMA-ES/pycma"
+
+    url = "https://github.com/CMA-ES/pycma/archive/refs/tags/r3.3.0.tar.gz"
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-pycma/package.py]: This package is on PyPi under the name cma

File: var/spack/repos/builtin/packages/py-pycma/package.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    maintainers("LydDeb")
+
+    version("3.3.0", sha256="2a061296bfca4eb979ed471e0c5acf40fc69345c13d4522ac6ed62a50d864bcf")
+
+    variant("plotting", default=True, description="Build with matplotlib support.")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-pycma/package.py]: Usually we disable variants by default (same as it is when you install with pip).

File: var/spack/repos/builtin/packages/py-pycma/package.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    variant(
+        "constrained_solution_tracking",
+        default=True,
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-pycma/package.py]: Same as above: disable by default.

File: var/spack/repos/builtin/packages/py-pycma/package.py (hunk 1 above)
Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-pycma/package.py]: You are right, I renamed the recipe and used pypi's sources.

File: var/spack/repos/builtin/packages/py-pycma/package.py (hunk 2 above)
Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-pycma/package.py]: Done

File: var/spack/repos/builtin/packages/py-pycma/package.py (hunk 3 above)
Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-pycma/package.py]: Done

**Review by manuelakuhn — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**